File: tmdb-core/src/main/java/dev/reuss/tmdb/common/review/ReviewAuthorDetails.java
package dev.reuss.tmdb.common.review;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Author details for a TMDB review.
 *
 * @param name       author display name
 * @param username   author username
 * @param avatarPath author avatar path
 * @param rating     author rating
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ReviewAuthorDetails(
        String name,

        String username,

        @JsonProperty("avatar_path")
        String avatarPath,

        Double rating
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/review/ReviewAuthorDetails.kt
package dev.reuss.tmdb.common.review

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Author details for a TMDB review.
 *
 * @property name       author display name
 * @property username   author username
 * @property avatarPath author avatar path
 * @property rating     author rating
 */
@JvmRecord
data class ReviewAuthorDetails(
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("username")
    val username: String?,
    @all:JsonProperty("avatar_path")
    val avatarPath: String?,
    @all:JsonProperty("rating")
    val rating: Double?,
) : TmdbModel
